In [617]:
# print all the outputs in a cell
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [618]:
%autosave 0

Autosave disabled


### Import Trump File

In [619]:
f = open('Trump.txt','r') 
#open the file in read mode

In [620]:
trump = f.read()  
#read the trump file

In [621]:
print(trump[:1000])  
#printing a sample data just to get a basic idea about cleaning the data

b'RT @Nettaaaaaaaa: Really wild how the insane things he said about Ferguson &amp; the death of Mike Brown Jr. were overlooked. https://t.co/R9Nd\xe2\x80\xa6\n'b'RT @Maria_squirt: @usaguy @JMoore821 @Havok_2018 @DanielBeerthuis @realDonaldTrump @LunaLuvgood2017 @StormResist @NatCookResists @DanaScott\xe2\x80\xa6\n'b'RT @NBCNews: Deputy AG Rosenstein directly approved application for search warrant on Trump personal attorney Michael Cohen, person with kn\xe2\x80\xa6\n'b'RT @HoarseWisperer: BWAHAHAHAHHAHAHAA.\n\nTrump fired Preet Bharara and personally interviewed and appointed his replacement...\n\n...and then\xe2\x80\xa6\n'b'RT @GreggJarrett: I\xe2\x80\x99ve written a book, soon to be published:  \xe2\x80\x9cThe Russia Hoax: The Illicit Scheme to Clear Hillary Clinton and Frame Donal\xe2\x80\xa6\n'b"@hughhewitt @realDonaldTrump Oh, puh-leeze, Hugh.\nTrump's personally interviewed and appointed AG signed off on thi\xe2\x80\xa6 https://t.co/vWBOXdNXP9\n"b'RT @eugenegu: What a powerful th

In [622]:
type(trump)
#to check the type 

str

### Raw tweets for word 'trump'

In [623]:
len(trump)

243459

### Cleaning HTTP Tag

In [624]:
import re 
result = re.sub(r"http\S+", "", trump)    
#Cleaning the http tag

In [625]:
print(result[:1000])

b'RT @Nettaaaaaaaa: Really wild how the insane things he said about Ferguson &amp; the death of Mike Brown Jr. were overlooked.  @Maria_squirt: @usaguy @JMoore821 @Havok_2018 @DanielBeerthuis @realDonaldTrump @LunaLuvgood2017 @StormResist @NatCookResists @DanaScott\xe2\x80\xa6\n'b'RT @NBCNews: Deputy AG Rosenstein directly approved application for search warrant on Trump personal attorney Michael Cohen, person with kn\xe2\x80\xa6\n'b'RT @HoarseWisperer: BWAHAHAHAHHAHAHAA.\n\nTrump fired Preet Bharara and personally interviewed and appointed his replacement...\n\n...and then\xe2\x80\xa6\n'b'RT @GreggJarrett: I\xe2\x80\x99ve written a book, soon to be published:  \xe2\x80\x9cThe Russia Hoax: The Illicit Scheme to Clear Hillary Clinton and Frame Donal\xe2\x80\xa6\n'b"@hughhewitt @realDonaldTrump Oh, puh-leeze, Hugh.\nTrump's personally interviewed and appointed AG signed off on thi\xe2\x80\xa6  @eugenegu: What a powerful thread on the true nature of American politics, the rise of Trump, t

### Cleaning embedded computer code with pattern of \xe2

In [626]:
result1= re.sub(r'(\\x(.){2})', '',result) 
#Cleaning the embedded code by computer

In [627]:
print(result1[:2000])

b'RT @Nettaaaaaaaa: Really wild how the insane things he said about Ferguson &amp; the death of Mike Brown Jr. were overlooked.  @Maria_squirt: @usaguy @JMoore821 @Havok_2018 @DanielBeerthuis @realDonaldTrump @LunaLuvgood2017 @StormResist @NatCookResists @DanaScott\n'b'RT @NBCNews: Deputy AG Rosenstein directly approved application for search warrant on Trump personal attorney Michael Cohen, person with kn\n'b'RT @HoarseWisperer: BWAHAHAHAHHAHAHAA.\n\nTrump fired Preet Bharara and personally interviewed and appointed his replacement...\n\n...and then\n'b'RT @GreggJarrett: Ive written a book, soon to be published:  The Russia Hoax: The Illicit Scheme to Clear Hillary Clinton and Frame Donal\n'b"@hughhewitt @realDonaldTrump Oh, puh-leeze, Hugh.\nTrump's personally interviewed and appointed AG signed off on thi  @eugenegu: What a powerful thread on the true nature of American politics, the rise of Trump, the root of white supremacy, and the solut\n'b'Queen Elizabeth likens Trump to noisy 

### Removing the retweets and &amp and \n

In [628]:
pun = ['RT','&amp','\\n']
for i in pun:
     if i in result1:
            result1 = result1.replace(i,'')
#removing the Retweets and &amp tags

print(result1[:2000])

b' @Nettaaaaaaaa: Really wild how the insane things he said about Ferguson ; the death of Mike Brown Jr. were overlooked.  @Maria_squirt: @usaguy @JMoore821 @Havok_2018 @DanielBeerthuis @realDonaldTrump @LunaLuvgood2017 @StormResist @NatCookResists @DanaScott'b' @NBCNews: Deputy AG Rosenstein directly approved application for search warrant on Trump personal attorney Michael Cohen, person with kn'b' @HoarseWisperer: BWAHAHAHAHHAHAHAA.Trump fired Preet Bharara and personally interviewed and appointed his replacement......and then'b' @GreggJarrett: Ive written a book, soon to be published:  The Russia Hoax: The Illicit Scheme to Clear Hillary Clinton and Frame Donal'b"@hughhewitt @realDonaldTrump Oh, puh-leeze, Hugh.Trump's personally interviewed and appointed AG signed off on thi  @eugenegu: What a powerful thread on the true nature of American politics, the rise of Trump, the root of white supremacy, and the solut'b'Queen Elizabeth likens Trump to noisy helicopter  @krassenstein: The c

### Removing all the punctuations

In [629]:
import string
x = str.maketrans("","",string.punctuation)
#Removing all the punctuations

In [630]:
result3 = result1.translate(x)    
#Result after cleaning the data

### Printing sample data after cleaning

In [631]:
print(result3[:1000]) 
#checking the cleaned data

b Nettaaaaaaaa Really wild how the insane things he said about Ferguson  the death of Mike Brown Jr were overlooked  Mariasquirt usaguy JMoore821 Havok2018 DanielBeerthuis realDonaldTrump LunaLuvgood2017 StormResist NatCookResists DanaScottb NBCNews Deputy AG Rosenstein directly approved application for search warrant on Trump personal attorney Michael Cohen person with knb HoarseWisperer BWAHAHAHAHHAHAHAATrump fired Preet Bharara and personally interviewed and appointed his replacementand thenb GreggJarrett Ive written a book soon to be published  The Russia Hoax The Illicit Scheme to Clear Hillary Clinton and Frame Donalbhughhewitt realDonaldTrump Oh puhleeze HughTrumps personally interviewed and appointed AG signed off on thi  eugenegu What a powerful thread on the true nature of American politics the rise of Trump the root of white supremacy and the solutbQueen Elizabeth likens Trump to noisy helicopter  krassenstein The chairman of Sinclair TV David Smith met with TrumpWe are here

### Length of data after cleaning

In [632]:
len(result3) #length of data after cleaning

184874

In [633]:
type(result3) #type of sample after data cleaning is still the same

str

### To convert every character to lower case and then split it into list

In [634]:
new = result3.lower().split()
#to convert every character to lower case and then split it into list

In [635]:
type(new) #checking the type of new

list

In [636]:
print(new[:100])

['b', 'nettaaaaaaaa', 'really', 'wild', 'how', 'the', 'insane', 'things', 'he', 'said', 'about', 'ferguson', 'the', 'death', 'of', 'mike', 'brown', 'jr', 'were', 'overlooked', 'mariasquirt', 'usaguy', 'jmoore821', 'havok2018', 'danielbeerthuis', 'realdonaldtrump', 'lunaluvgood2017', 'stormresist', 'natcookresists', 'danascottb', 'nbcnews', 'deputy', 'ag', 'rosenstein', 'directly', 'approved', 'application', 'for', 'search', 'warrant', 'on', 'trump', 'personal', 'attorney', 'michael', 'cohen', 'person', 'with', 'knb', 'hoarsewisperer', 'bwahahahahhahahaatrump', 'fired', 'preet', 'bharara', 'and', 'personally', 'interviewed', 'and', 'appointed', 'his', 'replacementand', 'thenb', 'greggjarrett', 'ive', 'written', 'a', 'book', 'soon', 'to', 'be', 'published', 'the', 'russia', 'hoax', 'the', 'illicit', 'scheme', 'to', 'clear', 'hillary', 'clinton', 'and', 'frame', 'donalbhughhewitt', 'realdonaldtrump', 'oh', 'puhleeze', 'hughtrumps', 'personally', 'interviewed', 'and', 'appointed', 'ag', 's

### Length of new list created

In [637]:
len(new) #length of new after list creation

29064

In [638]:
print(new[:100]) #printing the sample of this list

['b', 'nettaaaaaaaa', 'really', 'wild', 'how', 'the', 'insane', 'things', 'he', 'said', 'about', 'ferguson', 'the', 'death', 'of', 'mike', 'brown', 'jr', 'were', 'overlooked', 'mariasquirt', 'usaguy', 'jmoore821', 'havok2018', 'danielbeerthuis', 'realdonaldtrump', 'lunaluvgood2017', 'stormresist', 'natcookresists', 'danascottb', 'nbcnews', 'deputy', 'ag', 'rosenstein', 'directly', 'approved', 'application', 'for', 'search', 'warrant', 'on', 'trump', 'personal', 'attorney', 'michael', 'cohen', 'person', 'with', 'knb', 'hoarsewisperer', 'bwahahahahhahahaatrump', 'fired', 'preet', 'bharara', 'and', 'personally', 'interviewed', 'and', 'appointed', 'his', 'replacementand', 'thenb', 'greggjarrett', 'ive', 'written', 'a', 'book', 'soon', 'to', 'be', 'published', 'the', 'russia', 'hoax', 'the', 'illicit', 'scheme', 'to', 'clear', 'hillary', 'clinton', 'and', 'frame', 'donalbhughhewitt', 'realdonaldtrump', 'oh', 'puhleeze', 'hughtrumps', 'personally', 'interviewed', 'and', 'appointed', 'ag', 's

### Import Positive words

In [639]:
f2 = open('positive.txt','r')

### To read positive word list, change every character to lower case and split it into list

In [640]:
pw = f2.read().lower().split()
#to read positive word list, change every character to lower case and split it into list

In [641]:
type(pw) 
#type of pw

list

### Length of positive words

In [642]:
len(pw)
#number of positive words in list

2006

### Sample of positive words

In [643]:
print(pw[:100])
#printing the sampled list

['a+', 'abound', 'abounds', 'abundance', 'abundant', 'accessable', 'accessible', 'acclaim', 'acclaimed', 'acclamation', 'accolade', 'accolades', 'accommodative', 'accomodative', 'accomplish', 'accomplished', 'accomplishment', 'accomplishments', 'accurate', 'accurately', 'achievable', 'achievement', 'achievements', 'achievible', 'acumen', 'adaptable', 'adaptive', 'adequate', 'adjustable', 'admirable', 'admirably', 'admiration', 'admire', 'admirer', 'admiring', 'admiringly', 'adorable', 'adore', 'adored', 'adorer', 'adoring', 'adoringly', 'adroit', 'adroitly', 'adulate', 'adulation', 'adulatory', 'advanced', 'advantage', 'advantageous', 'advantageously', 'advantages', 'adventuresome', 'adventurous', 'advocate', 'advocated', 'advocates', 'affability', 'affable', 'affably', 'affectation', 'affection', 'affectionate', 'affinity', 'affirm', 'affirmation', 'affirmative', 'affluence', 'affluent', 'afford', 'affordable', 'affordably', 'afordable', 'agile', 'agilely', 'agility', 'agreeable', 'ag

### Import Negative words

In [644]:
f3 = open('neg.txt','r',encoding = "ISO-8859-1")
#opening the file with negative word, since was in another format so converted to ISO format for compatibility

### Read the negative word list, changed it to lower case and then split them to list

In [645]:
nw = f3.read().lower().split()

In [646]:
type(nw) 
#to check the type

list

### Length of negative words

In [647]:
len(nw)
#to check the length

4783

### Sample of negative words

In [648]:
print(nw[:100])
#to print the sample of negative words

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable', 'abominably', 'abominate', 'abomination', 'abort', 'aborted', 'aborts', 'abrade', 'abrasive', 'abrupt', 'abruptly', 'abscond', 'absence', 'absent-minded', 'absentee', 'absurd', 'absurdity', 'absurdly', 'absurdness', 'abuse', 'abused', 'abuses', 'abusive', 'abysmal', 'abysmally', 'abyss', 'accidental', 'accost', 'accursed', 'accusation', 'accusations', 'accuse', 'accuses', 'accusing', 'accusingly', 'acerbate', 'acerbic', 'acerbically', 'ache', 'ached', 'aches', 'achey', 'aching', 'acrid', 'acridly', 'acridness', 'acrimonious', 'acrimoniously', 'acrimony', 'adamant', 'adamantly', 'addict', 'addicted', 'addicting', 'addicts', 'admonish', 'admonisher', 'admonishingly', 'admonishment', 'admonition', 'adulterate', 'adulterated', 'adulteration', 'adulterier', 'adversarial', 'adversary', 'adverse', 'adversity', 'afflict', 'affliction', 'afflictive', 'affront', 'afraid', 'aggravate', 'aggravating', 'aggravation', 'aggression', 'aggress

### Import stop words

In [649]:
f4 = open('stopwords.txt','r')

### Read the stop word list, changed it to lower case and then split them to list

In [650]:
sw = f4.read().lower().split()

### Length of Stop words

In [651]:
len(sw)
#length of stop words

423

### Sample of Stop words

In [652]:
print(sw[:100])
#printing the stop words

['a', 'about', 'above', 'across', 'after', 'again', 'against', 'all', 'almost', 'alone', 'along', 'already', 'also', 'although', 'always', 'among', 'an', 'and', 'another', 'any', 'anybody', 'anyone', 'anything', 'anywhere', 'are', 'area', 'areas', 'around', 'as', 'ask', 'asked', 'asking', 'asks', 'at', 'away', 'b', 'back', 'backed', 'backing', 'backs', 'be', 'became', 'because', 'become', 'becomes', 'been', 'before', 'began', 'behind', 'being', 'beings', 'best', 'better', 'between', 'big', 'both', 'but', 'by', 'c', 'came', 'can', 'cannot', 'case', 'cases', 'certain', 'certainly', 'clear', 'clearly', 'come', 'could', 'd', 'did', 'differ', 'different', 'differently', 'do', 'does', 'done', 'down', 'downed', 'downing', 'downs', 'during', 'e', 'each', 'early', 'either', 'end', 'ended', 'ending', 'ends', 'enough', 'even', 'evenly', 'ever', 'every', 'everybody', 'everyone', 'everything', 'everywhere']


### Count the positive, negative, stop and other words

In [653]:
pscore = 0 #to initialize the positive score variable
p1 = []    #to create a list
nscore = 0
n1 = []
sscore = 0
s1 = []
oscore = 0
o1 = []

In [654]:
for i in new:
    if i !='trump':   #since trump can be a name so removing all the occurences of it as it might get counted in positive words
        if i in pw: 
            pscore += 1
            p1.append(i)
        elif i in nw:
            nscore -=1
            n1.append(i)
        elif i in sw:
            sscore +=1
            s1.append(i)
        else:
            oscore +=1
            o1.append(i)       

### Total positive word count in data

In [655]:
print(pscore) #total positive score 

623


### Total Negative word count in data

In [656]:
print(nscore) #total negative score

-886


### Conclusion: 
<b>The sum of positive words: 623 and the sum of negative words is: -886. The general sentiment is negative. For determining weakly or strongly negative, I have used ratio as below: </b>

In [657]:
ratio = abs(pscore/nscore)

In [658]:
ratio

0.7031602708803611

<b> I have defined the ratio here as: 
    <li>if abs(ratio)  >=0 and < 0.5 -> strongly negative </li>
    <li>abs(ratio) >= 0.5 and  < 1 -> weakly negative</li>
    <li>abs(ratio) >=1 and < 1.5 -> weakly positive</li>
    <li>abs(ratio) >=1.5  -> strongly positive</li>
      Since ratio is 0.7029478458049887.Based on the analysis here I think that the sentiment is weakly negative.</b>

In [669]:
print(oscore) #score of words which are neither positive nor negative or stop words

14725


In [670]:
print(sscore) #total stop words

11953


In [671]:
print(p1[:100]) #displaying the positive words

['clear', 'powerful', 'supremacy', 'luxury', 'supporter', 'winning', 'grand', 'master', 'grand', 'supporting', 'admire', 'top', 'protect', 'right', 'support', 'support', 'like', 'like', 'supports', 'great', 'great', 'well', 'obsession', 'leading', 'protection', 'right', 'privilege', 'loves', 'winning', 'like', 'well', 'like', 'like', 'whoa', 'poised', 'credible', 'strong', 'joy', 'worth', 'grand', 'worth', 'trust', 'faith', 'win', 'favor', 'like', 'right', 'privilege', 'wins', 'supreme', 'stunning', 'win', 'love', 'privilege', 'fun', 'helped', 'winning', 'well', 'like', 'happy', 'led', 'thank', 'winning', 'useful', 'privilege', 'stunning', 'victory', 'hero', 'perfectly', 'perfect', 'right', 'fair', 'supporting', 'top', 'loves', 'winning', 'work', 'privilege', 'enough', 'enough', 'success', 'great', 'thank', 'love', 'supporting', 'amazed', 'advantage', 'wonder', 'supreme', 'like', 'confidence', 'winning', 'important', 'love', 'like', 'liked', 'fine', 'protect', 'perfect', 'worth']


In [672]:
print(n1[:100]) #displaying the negative words

['wild', 'insane', 'death', 'hoax', 'illicit', 'noisy', 'dispute', 'stormy', 'hysteria', 'unbelievable', 'disgraceful', 'attack', 'rumor', 'fraudulent', 'hell', 'wild', 'terror', 'lost', 'attacks', 'lying', 'fraud', 'corruption', 'noisy', 'guilty', 'complacent', 'inevitably', 'terrorism', 'spew', 'denying', 'stormy', 'wild', 'insane', 'death', 'threatening', 'bugging', 'biased', 'furious', 'crap', 'damn', 'break', 'disgraceful', 'attack', 'killed', 'disagree', 'overwhelming', 'dense', 'condemns', 'heinous', 'attack', 'involuntary', 'criminal', 'thug', 'hell', 'obsess', 'lies', 'breaking', 'resignation', 'embattled', 'unsafe', 'destroy', 'fake', 'criminal', 'hard', 'dispute', 'problem', 'subpoena', 'dump', 'dirty', 'dangerous', 'hard', 'outrageous', 'attacks', 'bigotry', 'condemns', 'heinous', 'attack', 'stormy', 'deny', 'worry', 'loses', 'stuck', 'criminal', 'corruption', 'refusing', 'lied', 'hell', 'crime', 'meltdown', 'doubt', 'crazy', 'explode', 'idiot', 'disgrace', 'issues', 'break

In [673]:
print(s1[:100]) #displaying the stopwords

['b', 'really', 'how', 'the', 'things', 'he', 'said', 'about', 'the', 'of', 'were', 'for', 'on', 'with', 'and', 'and', 'his', 'a', 'to', 'be', 'the', 'the', 'to', 'and', 'and', 'off', 'on', 'what', 'a', 'on', 'the', 'of', 'the', 'of', 'the', 'of', 'and', 'the', 'to', 'the', 'of', 'with', 'are', 'here', 'to', 'your', 'the', 'for', 'in', 'from', 'its', 'in', 'a', 'over', 'of', 'a', 'that', 'we', 'in', 'a', 'of', 'from', 'us', 'of', 'the', 'of', 'new', 'a', 'gets', 'the', 'and', 'seconds', 'later', 'in', 'over', 'his', 'to', 'for', 'one', 'in', 'to', 'how', 'your', 'was', 'in', 'an', 'by', 'new', 'the', 'a', 'in', 'the', 'are', 'said', 'to', 'have', 'with', 'who', 'gives', 'a']


In [674]:
print(o1[:100]) #displaying the other words

['nettaaaaaaaa', 'ferguson', 'mike', 'brown', 'jr', 'overlooked', 'mariasquirt', 'usaguy', 'jmoore821', 'havok2018', 'danielbeerthuis', 'realdonaldtrump', 'lunaluvgood2017', 'stormresist', 'natcookresists', 'danascottb', 'nbcnews', 'deputy', 'ag', 'rosenstein', 'directly', 'approved', 'application', 'search', 'warrant', 'personal', 'attorney', 'michael', 'cohen', 'person', 'knb', 'hoarsewisperer', 'bwahahahahhahahaatrump', 'fired', 'preet', 'bharara', 'personally', 'interviewed', 'appointed', 'replacementand', 'thenb', 'greggjarrett', 'ive', 'written', 'book', 'soon', 'published', 'russia', 'scheme', 'hillary', 'clinton', 'frame', 'donalbhughhewitt', 'realdonaldtrump', 'oh', 'puhleeze', 'hughtrumps', 'personally', 'interviewed', 'appointed', 'ag', 'signed', 'thi', 'eugenegu', 'thread', 'true', 'nature', 'american', 'politics', 'rise', 'root', 'white', 'solutbqueen', 'elizabeth', 'likens', 'helicopter', 'krassenstein', 'chairman', 'sinclair', 'tv', 'david', 'smith', 'met', 'trumpwe', 'd

### Total count of all words

In [675]:
total = pscore + abs(nscore) + sscore + oscore #total count of all words
total

28187

#### ratio of positive word to the whole word count

In [676]:
ratiop = pscore/total #ratio of positive word to the whole word count

#### Ratio of negative word to the whole word count

In [677]:
ration = abs(nscore)/total #ratio of negative word to the whole word count

#### Ratio of Stop word to the whole word count

In [678]:
ratios = sscore/total #ratio of stop word to the whole word count

#### Ratio of other word to the whole word count

In [679]:
ratioo = oscore/total #ratio of other word to the whole word count

In [680]:
round(ratiop,4)  #rounded positive ratio to 4 decimal places

0.0221

In [681]:
round(ration,4) #rounded negative ratio to 4 decimal places

0.0314

In [682]:
round(ratios,4) #rounded stopword ratio to 4 decimal places

0.4241

In [683]:
round(ratioo,4) #rounded other ratio to 4 decimal places

0.5224

In [684]:
f.close()

In [685]:
f2.close()

In [686]:
f3.close()

In [687]:
f4.close()

## Second word Analysis

### Analyzing the tweet for NationalPetDay

In [688]:
nf = open('NationalPetDay.txt','r')

In [689]:
pet = nf.read()

### Sample raw tweets

In [690]:
print(pet[:1000])

b"RT @fusetv: .@ArianaGrande, @BritneySpears, @BTS_twt's V + 16 more celebs with their super cute pets \xf0\x9f\x98\x8d \xe2\x80\x93 https://t.co/F4RQ0rxnIa #NationalP\xe2\x80\xa6\n"b"RT @firtsIove: happy #NationalPetDay to yoongi's puppy \xe2\x99\xa1 https://t.co/3EPLVv7Rhs\n"b"RT @fusetv: .@ArianaGrande, @BritneySpears, @BTS_twt's V + 16 more celebs with their super cute pets \xf0\x9f\x98\x8d \xe2\x80\x93 https://t.co/F4RQ0rxnIa #NationalP\xe2\x80\xa6\n"b'RT @icklenellierose: Happy #NationalPetDay https://t.co/1Pw5tnTB9O\n'b"RT @DeptofDefense: It\xe2\x80\x99s #NationalPetDay \xf0\x9f\x90\xb6 and even though Chesty XV is just a puppy \xf0\x9f\x90\xbe, he's no pet because he is becoming a Marine, one com\xe2\x80\xa6\n"b'someone is quite excited it\xe2\x80\x99s #nationalpetday https://t.co/vClUTv3xC6\n'b'RT @Jeep: Fluff riders. #NationalPetDay https://t.co/gsgnLv69Q0\n'b"RT @fusetv: .@ArianaGrande, @BritneySpears, @BTS_twt's V + 16 more celebs with their super cute pets \xf0\x9f\x98\x8d

In [691]:
type(pet)
#to check the type 

str

### Length of raw tweets for NationalPetDay

In [692]:
len(pet)

41546

### Cleaning HTTP Tag

In [693]:
import re
pet1 = re.sub(r"http\S+", "", pet)    
#Cleaning the http tag

### Cleaning embedded computer code with patter of \xe2\x80\etc

In [694]:
pet2= re.sub(r'(\\x(.){2})', '',pet1) 
#Cleaning the embedded code by computer

### Removing the retweets and &amp, \n

In [695]:
punc = ['RT','&amp','\\n']
for j in punc:
     if j in pet2:
            pet2 = pet2.replace(j,'')
#removing the Retweets and &amp tags

print(pet2[:1000])

b" @fusetv: .@ArianaGrande, @BritneySpears, @BTS_twt's V + 16 more celebs with their super cute pets    #NationalP"b" @firtsIove: happy #NationalPetDay to yoongi's puppy   @fusetv: .@ArianaGrande, @BritneySpears, @BTS_twt's V + 16 more celebs with their super cute pets    #NationalP"b' @icklenellierose: Happy #NationalPetDay  @DeptofDefense: Its #NationalPetDay  and even though Chesty XV is just a puppy , he's no pet because he is becoming a Marine, one com"b'someone is quite excited its #nationalpetday  @Jeep: Fluff riders. #NationalPetDay  @fusetv: .@ArianaGrande, @BritneySpears, @BTS_twt's V + 16 more celebs with their super cute pets    #NationalP"b' @ChrisEvans: This is the moment we met. He was trying so hard to stay seated even though he desperately wanted to get out. I knew right'b" @danicamckellar: It's #NationalPetDay! And the @americanhumane @herodogawards are the Oscars for dogs: Pick your favorite #HeroDogs for"b" @CMHSPets: Happy #NationalPetDay! We're so excited to annou

### Removing all the punctuations

In [696]:
import string
y = str.maketrans("","",string.punctuation)

In [697]:
pet3 = pet2.translate(y)    
#Result after cleaning the data

### Printing sample data after cleaning

In [698]:
print(pet3[:1000]) 
#checking the cleaned data

b fusetv ArianaGrande BritneySpears BTStwts V  16 more celebs with their super cute pets    NationalPb firtsIove happy NationalPetDay to yoongis puppy   fusetv ArianaGrande BritneySpears BTStwts V  16 more celebs with their super cute pets    NationalPb icklenellierose Happy NationalPetDay  DeptofDefense Its NationalPetDay  and even though Chesty XV is just a puppy  hes no pet because he is becoming a Marine one combsomeone is quite excited its nationalpetday  Jeep Fluff riders NationalPetDay  fusetv ArianaGrande BritneySpears BTStwts V  16 more celebs with their super cute pets    NationalPb ChrisEvans This is the moment we met He was trying so hard to stay seated even though he desperately wanted to get out I knew rightb danicamckellar Its NationalPetDay And the americanhumane herodogawards are the Oscars for dogs Pick your favorite HeroDogs forb CMHSPets Happy NationalPetDay Were so excited to announce our Spring Fling Adoption Event happening this weekend April 1315 RedubHappy Nati

### Length of data after cleaning

In [699]:
len(pet3) 

26709

In [700]:
type(pet3)

str

### To convert every character to lower case and then split it into list

In [701]:
newpet = pet3.lower().split()
#to convert every character to lower case and then split it into list

In [702]:
type(newpet) #checking the type of new

list

In [703]:
print(newpet[:100]) #printing the sample of this list

['b', 'fusetv', 'arianagrande', 'britneyspears', 'btstwts', 'v', '16', 'more', 'celebs', 'with', 'their', 'super', 'cute', 'pets', 'nationalpb', 'firtsiove', 'happy', 'nationalpetday', 'to', 'yoongis', 'puppy', 'fusetv', 'arianagrande', 'britneyspears', 'btstwts', 'v', '16', 'more', 'celebs', 'with', 'their', 'super', 'cute', 'pets', 'nationalpb', 'icklenellierose', 'happy', 'nationalpetday', 'deptofdefense', 'its', 'nationalpetday', 'and', 'even', 'though', 'chesty', 'xv', 'is', 'just', 'a', 'puppy', 'hes', 'no', 'pet', 'because', 'he', 'is', 'becoming', 'a', 'marine', 'one', 'combsomeone', 'is', 'quite', 'excited', 'its', 'nationalpetday', 'jeep', 'fluff', 'riders', 'nationalpetday', 'fusetv', 'arianagrande', 'britneyspears', 'btstwts', 'v', '16', 'more', 'celebs', 'with', 'their', 'super', 'cute', 'pets', 'nationalpb', 'chrisevans', 'this', 'is', 'the', 'moment', 'we', 'met', 'he', 'was', 'trying', 'so', 'hard', 'to', 'stay', 'seated', 'even']


### Length of new list created

In [704]:
len(newpet) #length of new after list creation

4320

### Import Positive words

In [705]:
nf2 = open('positive.txt','r')

### To read positive word list, change every character to lower case and split it into list

In [706]:
pwt = nf2.read().lower().split()
#to read positive word list, change every character to lower case and split it into list

In [707]:
type(pwt) 
#type of pw

list

### Length of positive words

In [708]:
len(pwt)
#number of positive words in list

2006

### Sample of positive words

In [709]:
print(pwt[:100])
#printing the sampled list

['a+', 'abound', 'abounds', 'abundance', 'abundant', 'accessable', 'accessible', 'acclaim', 'acclaimed', 'acclamation', 'accolade', 'accolades', 'accommodative', 'accomodative', 'accomplish', 'accomplished', 'accomplishment', 'accomplishments', 'accurate', 'accurately', 'achievable', 'achievement', 'achievements', 'achievible', 'acumen', 'adaptable', 'adaptive', 'adequate', 'adjustable', 'admirable', 'admirably', 'admiration', 'admire', 'admirer', 'admiring', 'admiringly', 'adorable', 'adore', 'adored', 'adorer', 'adoring', 'adoringly', 'adroit', 'adroitly', 'adulate', 'adulation', 'adulatory', 'advanced', 'advantage', 'advantageous', 'advantageously', 'advantages', 'adventuresome', 'adventurous', 'advocate', 'advocated', 'advocates', 'affability', 'affable', 'affably', 'affectation', 'affection', 'affectionate', 'affinity', 'affirm', 'affirmation', 'affirmative', 'affluence', 'affluent', 'afford', 'affordable', 'affordably', 'afordable', 'agile', 'agilely', 'agility', 'agreeable', 'ag

### Import Negative words

In [710]:
nf3 = open('neg.txt','r',encoding = "ISO-8859-1")
#opening the file with negative word, since was in another format so converted to ISO format for compatibility

### Read the negative word list, changed it to lower case and then split them to list

In [711]:
nwt = nf3.read().lower().split()
#read the negative word list, changed it to lower case and then split them to list

In [712]:
type(nwt) 
#to check the type

list

### Length of negative words

In [713]:
len(nwt)
#to check the length

4783

### Sample of negative words

In [714]:
print(nwt[:100])
#to print the sample of negative words

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable', 'abominably', 'abominate', 'abomination', 'abort', 'aborted', 'aborts', 'abrade', 'abrasive', 'abrupt', 'abruptly', 'abscond', 'absence', 'absent-minded', 'absentee', 'absurd', 'absurdity', 'absurdly', 'absurdness', 'abuse', 'abused', 'abuses', 'abusive', 'abysmal', 'abysmally', 'abyss', 'accidental', 'accost', 'accursed', 'accusation', 'accusations', 'accuse', 'accuses', 'accusing', 'accusingly', 'acerbate', 'acerbic', 'acerbically', 'ache', 'ached', 'aches', 'achey', 'aching', 'acrid', 'acridly', 'acridness', 'acrimonious', 'acrimoniously', 'acrimony', 'adamant', 'adamantly', 'addict', 'addicted', 'addicting', 'addicts', 'admonish', 'admonisher', 'admonishingly', 'admonishment', 'admonition', 'adulterate', 'adulterated', 'adulteration', 'adulterier', 'adversarial', 'adversary', 'adverse', 'adversity', 'afflict', 'affliction', 'afflictive', 'affront', 'afraid', 'aggravate', 'aggravating', 'aggravation', 'aggression', 'aggress

### Import stop words


In [715]:
nf4 = open('stopwords.txt','r')

### Read the stop word list, changed it to lower case and then split them to list

In [716]:
swt = nf4.read().lower().split()

### Length of Stop words

In [717]:
len(swt)
#length of stop words

423

### Sample of Stop words

In [718]:
print(swt[:100])
#printing the stop words

['a', 'about', 'above', 'across', 'after', 'again', 'against', 'all', 'almost', 'alone', 'along', 'already', 'also', 'although', 'always', 'among', 'an', 'and', 'another', 'any', 'anybody', 'anyone', 'anything', 'anywhere', 'are', 'area', 'areas', 'around', 'as', 'ask', 'asked', 'asking', 'asks', 'at', 'away', 'b', 'back', 'backed', 'backing', 'backs', 'be', 'became', 'because', 'become', 'becomes', 'been', 'before', 'began', 'behind', 'being', 'beings', 'best', 'better', 'between', 'big', 'both', 'but', 'by', 'c', 'came', 'can', 'cannot', 'case', 'cases', 'certain', 'certainly', 'clear', 'clearly', 'come', 'could', 'd', 'did', 'differ', 'different', 'differently', 'do', 'does', 'done', 'down', 'downed', 'downing', 'downs', 'during', 'e', 'each', 'early', 'either', 'end', 'ended', 'ending', 'ends', 'enough', 'even', 'evenly', 'ever', 'every', 'everybody', 'everyone', 'everything', 'everywhere']


### Count the positive, negative, stop and other words

In [719]:
petpscore = 0 #to initialize the positive score variable
pp1 = []    #to create a list
petnscore = 0
np1 = []
petsscore = 0
sp1 = []
petoscore = 0
op1 = []

In [720]:
for i in newpet: 
    if i in pwt: 
        petpscore += 1
        pp1.append(i)
    elif i in nwt:
        petnscore -=1
        np1.append(i)
    elif i in swt:
        petsscore +=1
        sp1.append(i)
    else:
        petoscore +=1
        op1.append(i)       

### Total positive word count in data

In [721]:
print(petpscore) #total positive score 

231


### Total Negative word count in data

In [722]:
print(petnscore)#total negative score

-126


### Total Stop word count in data

In [723]:
print(petsscore)

2112


### Total other word count in data

In [724]:
print(petoscore)

1851


### Conclusion:  From the above code, out of total 4320 words, 231 are positive and 126 are negative, 2112 are stop words. The general sentiment of people for National Pet Day is positive. Now, to determine weakly or strongly, lets calculate ratio

In [725]:
testratio= abs(petpscore/petnscore)
testratio

1.8333333333333333

<b> I have defined the ratio here as: 
    <li> I defined:</li>
    <li>if abs(ratio)  >=0 and < 0.5 -> strongly negative </li>
    <li>abs(ratio) >= 0.5 and < 1 -> weakly negative</li>
    <li>abs(ratio) >=1 and < 1.5 -> weakly positive</li>
    <li>abs(ratio) >=1.5  -> strongly positive</li></b>
    
###  Since ratio is 1.83 Based on the analysis here I think that the sentiment is strongly positive

In [726]:
sumpet = petpscore + abs(petnscore) + petsscore + petoscore
sumpet

4320

In [727]:
pratio = petpscore/sumpet
pratio

0.05347222222222222

In [728]:
nratio = abs(petnscore)/sumpet
nratio

0.029166666666666667

In [729]:
sratio = petsscore/sumpet
sratio

0.4888888888888889

In [730]:
oratio = petoscore/sumpet
oratio

0.4284722222222222

In [731]:
nf.close()

In [732]:
nf2.close()

In [733]:
nf3.close()

In [551]:
nf4.close()